src/uu/cp/src/cp.rs

```diff
@@ -1134,7 +1135,24 @@ fn parse_path_args(
         }
     }
 
-    Ok((paths, target))
+    // If the path exists and is valid keep it in the argument list.
+    // Exception for symlinks as they are a bit more tricky to handle
+    // and there's logic further down the execution to handle them.
```

Comment:
> there's logic further down

So this code doesn't actually handle the error as it happens, but rather tries to predict whether an error is about to happen? I don't like it; this type of approach is inevitably fragile: What if we check a file now, but later it appears? When copying several gigabytes on a slow disk, this might be a window of minutes! There are several options that change behavior around symlinks: Either this code results in the wrong error message for those, or wrong behavior, or both. If there are any other problems when copying, I get the impression that GNU cp would continue with the next file, and uutils would still abort; i.e. the same issue that this PR is trying to fix, but in a slightly different flavor.

No. Instead, let's handle the error if and when it happens. This should also have the advantage of reducing the TOCTOU-timespan, and guaranteed robustness: There cannot be false-positive or false-negatives when there is no "prediction" involved.